platforms/core-configuration/model-core/src/main/java/org/gradle/api/internal/provider/AbstractCollectionProperty.java

```diff
@@ -77,7 +83,7 @@
  * @param <C> the type of {@link Collection} (as returned by {@link ProviderInternal#getType()}) that corresponds to this collection property's realized value, for instance, when {@link Provider#get()} is invoked.
  */
 public abstract class AbstractCollectionProperty<T, C extends Collection<T>> extends AbstractProperty<C, CollectionSupplier<T, C>>
-    implements CollectionPropertyInternal<T, C> {
+    implements CollectionPropertyInternal<T, C>, SupportsCompoundAssignment<AbstractCollectionProperty<T, C>.CompoundAssignmentStandIn> {
```

Comment:
🤔 We've been working hard to avoid Kotlinisms to leak into our implementations and here we are accepting Groovyisms again... Hard call.